project-base/storefront/hooks/cart/useRemoveFromCart.ts

```diff
@@ -1,26 +1,36 @@
-import { CartFragmentApi, CartItemFragmentApi, useRemoveFromCartMutationApi } from 'graphql/generated';
+import { CartItemFragmentApi, useCartQueryApi, useRemoveFromCartMutationApi } from 'graphql/generated';
 import { onGtmRemoveFromCartEventHandler } from 'gtm/helpers/eventHandlers';
 import { GtmProductListNameType } from 'gtm/types/enums';
 import { mapPriceForCalculations } from 'helpers/mappers/price';
+import { useIsUserLoggedIn } from 'hooks/auth/useIsUserLoggedIn';
 import { useDomainConfig } from 'hooks/useDomainConfig';
 import { usePersistStore } from 'store/usePersistStore';
 
-export type RemoveFromCartHandler = (
-    cartItem: CartItemFragmentApi,
-    listIndex: number,
-) => Promise<CartFragmentApi | undefined | null>;
+export type RemoveFromCartHandler = (cartItem: CartItemFragmentApi, listIndex: number) => Promise<void>;
 
 export const useRemoveFromCart = (gtmProductListName: GtmProductListNameType): [RemoveFromCartHandler, boolean] => {
     const [{ fetching }, removeItemFromCart] = useRemoveFromCartMutationApi();
     const { url, currencyCode } = useDomainConfig();
     const cartUuid = usePersistStore((store) => store.cartUuid);
+    const isUserLoggedIn = useIsUserLoggedIn();
+
+    const [, refetchCart] = useCartQueryApi({
+        variables: { cartUuid },
+        pause: !cartUuid && !isUserLoggedIn,
+        requestPolicy: 'network-only',
+    });
+
     const updateCartUuid = usePersistStore((store) => store.updateCartUuid);
 
     const removeItemFromCartAction = async (cartItem: CartItemFragmentApi, listIndex: number) => {
         const removeItemFromCartActionResult = await removeItemFromCart({
             input: { cartUuid, cartItemUuid: cartItem.uuid },
         });
 
+        if (removeItemFromCartActionResult.error) {
+            refetchCart();
+        }
```

Comment:
You can test it on your own. This implementation I would consider it as a bugfix. When you add to cart some product, then open another tab, remove the product from cart, go to other tab and try to remove it there as well. It won't do anything. Because of that it make sense to refetch cart so user get the fresh results.

Reply:
Okay. Should we apply this logic to other cart mutations as well? Just a thought, you can resolve this thread as you wish...

Reply:
I haven't found another case where this would be needed. In case of Add to cart it will just add one more to quantity and return correct result. In case of changing quantity, it's the same result. If you have some improvements on your mind, we can create separate task. This is just something which was sort of connected to this synchronization.